src/main/java/UI/Presenter/HeaderPresenter.java

```diff
@@ -91,5 +92,38 @@ public void registerUser(String username, String email, String password) {
     public void logoutUser(){
         
     }
+
+    public void openNewShop(String shopName, String bankDetails, String shopAddress){
+        RestTemplate restTemplate = new RestTemplate();
+        ShopDto shopDto = new ShopDto(shopName, bankDetails, shopAddress);
+
+        UI.getCurrent().getPage().executeJs("return localStorage.getItem('authToken');")
+                .then(String.class, token -> {
+                    if (token != null && !token.isEmpty()) {
+                        System.out.println("Token: " + token);
+
+                        HttpHeaders headers = new HttpHeaders();
+                        headers.add("Authorization", token);
+
+                        HttpEntity<ShopDto> requestEntity = new HttpEntity<>(shopDto, headers);
+
+                        ResponseEntity<String> response = restTemplate.exchange(
+                                "http://localhost:" + _serverPort + "/api/shop/openNewShop",
+                                HttpMethod.POST,
+                                requestEntity,
+                                String.class);
+
+                        if (response.getStatusCode().is2xxSuccessful()) {
```

Comment:
pls see the new commit Amit is making, keep in mind there will be some changes here